base/collection_xattr.go

```diff
@@ -677,3 +708,15 @@ func gocbMutationMacro(meType sgbucket.MacroExpansionType) gocb.MutationMacro {
 		return gocb.MutationMacroCAS
 	}
 }
+
+// getUpsertSpecs returns a slice of gocb.MutateInSpec for the given xattrs, or returns an error if any values are nil.
+func getUpsertSpecs(xattrs map[string][]byte) ([]gocb.MutateInSpec, error) {
```

Comment:
```suggestion
func getUpsertSpecsForXattrs(xattrs map[string][]byte) ([]gocb.MutateInSpec, error) {
```
This might look like nitpicking but I found the generic "getUpsertSpecs" to be very ambiguous.